codegra_fs/cgfs.py

```diff
@@ -1621,9 +1907,109 @@ def write(self, path, data, offset, fh):
+
+    if not fixed:
+        logger.warning('=====================================================')
+        logger.warning('Mounting in non-fixed mode, all changes will be')
+        logger.warning('visible and additions to students.')
+        logger.warning('Watch out when uploading grading scripts!')
+        logger.warning('=====================================================')
+
+    with tempfile.TemporaryDirectory(dir=tempfile.gettempdir()) as tmpdir:
+        sockfile = tempfile.NamedTemporaryFile().name
+        kwargs: t.Dict[str, str] = {}
+        if sys.platform.startswith('darwin'):
+            # Fix OSX encoding issue as described here:
+            # https://web.archive.org/web/20180920131107/https://github.com/osxfuse/osxfuse/issues/71
+            kwargs = {
+                'from_code': 'UTF-8',
+                'to_code': 'UTF-8-MAC',
+                'modules': 'iconv',
+            }
+
+        fs = None
+        try:
+            fs = CGFS(
+                latest_only=latest_only,
+                socketfile=sockfile,
+                fixed=fixed,
+                mountpoint=mountpoint,
+                tmpdir=tmpdir,
+                rubric_append_only=rubric_append_only,
+                assigned_only=assigned_only,
+            )
+            FUSE(
+                fs,
+                mountpoint,
+                nothreads=True,
+                foreground=True,
+                direct_io=True,
+                **kwargs,
+            )
+        except RuntimeError:  # pragma: no cover
+            logger.critical(traceback.format_exc())
+        finally:
+            if fs is not None and hasattr(fs, 'api_handler'):
+                fs.api_handler.stop = True
+            if os.path.isfile(sockfile):
+                os.unlink(sockfile)
+
+
+def newer_version_available() -> bool:
+    req = requests.get('https://codegra.de/.cgfs.version', timeout=2)
+    return req.status_code < 300 and tuple(
+        int(p) for p in req.content.decode('utf8').strip().split('.')
+    ) > codegra_fs.__version__
+
+
+def check_version() -> None:
+    if newer_version_available():
+        msg = [
+            (
+                'You are running an outdated version of'
+                ' CGFS, please consider upgrading.'
+            ),
+            'You can do this at https://codegra.de/codegra_fs/latest/',
+        ]
+        print('-' * (max(len(l) for l in msg) + 4), file=sys.stderr)
+        for line in msg:
+            print('| {} |'.format(line), file=sys.stderr)
+        print('-' * (max(len(l) for l in msg) + 4), file=sys.stderr)
+
+
+def main() -> None:
     global cgapi
 
+    msg = codegra_fs.utils.get_fuse_install_message()
+    if msg:
+        err, url = msg
+        print('ERROR!')
+        print(msg, file=sys.stderr)
+        if url:
+            print('You can download it here: {}'.format(url))
+        sys.exit(2)
+
+    check_version()
+
     argparser = ArgumentParser(description='CodeGra.de file system')
```

Comment:
Though not conventional, maybe we could add an [`epilog`](https://docs.python.org/3/library/argparse.html#epilog) describing the environment variables. We don't have a man page where they would usually be listed, but they're super useful.

Reply:
Yeah, that is probably a good idea.

Reply:
Sure.

Reply:
Maybe it is better to it here, you can decide. What should it contain?

Reply:
I was thinking something very simple along the lines of

```
Environment variables
CGAPI_BASE_URL      Base URL of your CodeGrade instance
CGFS_PASSWORD       Your CodeGrade password
```

It could be a little more verbose but not much I'd say.

Reply:
Ok seems fine, I will add it!